lib/active_merchant/billing/base.rb

```diff
@@ -1,32 +1,21 @@
 module ActiveMerchant #:nodoc:
   module Billing #:nodoc:
     module Base
-      # Set ActiveMerchant gateways in test mode.
-      #
-      #   ActiveMerchant::Billing::Base.gateway_mode = :test
-      mattr_accessor :gateway_mode
+      GATEWAY_MODE_DEPRECATION_MESSAGE = 'Base#gateway_mode is deprecated in favor of Base#mode and will be removed in a future version'
 
-      # Set ActiveMerchant integrations in test mode.
+      # Set ActiveMerchant gateways in test mode.
       #
-      #   ActiveMerchant::Billing::Base.integration_mode = :test
-      def self.integration_mode=(mode)
```

Comment:
The method doesn't work anyways. `OFFSITE_PAYMENT_EXTRACTION_MESSAGE` is undefined.